pkg/kev/config/schema.go

```diff
@@ -67,6 +67,82 @@ var ServicesSchema = map[string]interface{}{
 		LabelWorkloadReplicas,
 		LabelWorkloadLivenessProbeType,
 	},
+	"allOf": []map[string]interface{}{
+		// Liveness probe
+		{
+			"oneOf": []map[string]interface{}{
+				{
+					"properties": map[string]interface{}{
+						LabelWorkloadLivenessProbeType: map[string]interface{}{
+							"const": "http",
+						},
+					},
+					"required": []string{LabelWorkloadLivenessProbeHTTPPath, LabelWorkloadLivenessProbeHTTPPort},
+				},
+				{
+					"properties": map[string]interface{}{
+						LabelWorkloadLivenessProbeType: map[string]interface{}{
+							"const": "tcp",
+						},
+					},
+					"required": []string{LabelWorkloadLivenessProbeTCPPort},
+				},
+				{
+					"properties": map[string]interface{}{
+						LabelWorkloadLivenessProbeType: map[string]interface{}{
+							"const": "command",
+						},
+					},
+					"required": []string{LabelWorkloadLivenessProbeCommand},
+				},
+				// This acts as a catch all when the required properties are not provided, should come in the end
+				// otherwise instead of matching that "command" type needs a properties the error says the value is not "none".
+				{
+					"properties": map[string]interface{}{
```

Comment:
@mangas @marcinc I'm curious about the following situation: 

... Changing the default `kev.workload.liveness-probe-type` from `command` to `none`
... On render

1 - Should we ignore the existing `kev.workload.liveness-probe-command` and render (without setting a probe)?

OR

2 - Should we error and instruct the user to remove `kev.workload.liveness-probe-command` as its no used/needed?

**Currently the code does (1)**

Reply:
I thought about this and the use case I came with is if I want to disable the problem I can just change to none. I don't think we should error on that since the extra parameter doesn't really matter. Otherwise I'd have to remove the whole thing and add it back. The other side of this argument is that we could comment. Also raises the question of whether or not we care about extra properties enough to error because of them? Does that make sense?

Reply:
That's a case I brought up before around a use case where one iterates on the app with liveness probe defined, then temporarily turns it off (e.g. when testing), to then turn it back on. 

The current behaviour does the (1) as you said above, however, `render` will respect it and won't generate the probe body. This also enables the user to turn the probe back on by setting type to `command` without having to redefine the command. 

The only slightly negative side effect of the use case above could be that when user decides to drop the probe entirely by setting its type to `none` AND at the same time forgets to remove the command itself, then it'll remain untouched in service configuration labels. But I think we can live with that.

Reply:
This can get very confusing fast - even in other scenarios.

You can start collecting a lot of cruft.

For example this override works/renders - we had an `http` probe that got swapped to a `command` probe.

```yaml
...
  db-test:
    labels:
      kev.workload.liveness-probe-command: '["CMD", "echo", "Define healthcheck command for service db-test"]'
      kev.workload.liveness-probe-type: command
      kev.workload.readiness-probe-http-path: /status
      kev.workload.readiness-probe-http-port: "8080"
      kev.workload.replicas: "1"
...
```

I guess a **Warning** can help here ... (follow up PR?).

Reply:
I'd agree here. There is potential here for leftover buildup. Either a Warning, or straight cleanup at the cost of slight inconvenience on user's part - as this will require providing the details to re-enabled probe again which personally I don't think is a much of a problem, and this use case shouldn't happen too often I don't think.

Reply:
https://github.com/appvia/kev/issues/414